### PR TITLE
migrate gitlab source to genreconciler and drop reconciler.Base

### PR DESCRIPTION
**Pull request by n3wscott**
## Proposed Changes

- migrate gitlab source to genreconciler 
- drop reconciler.Base

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: gitlab/pkg/reconciler/controller.go [hunk 1]
```diff
@@ -18,7 +18,9 @@ package gitlab
 
 import (
 	"context"
```
Comment (mattmoor) [inline gitlab/pkg/reconciler/controller.go]: Format Go code:
```suggestion
	"context"

```

File: gitlab/pkg/reconciler/controller.go [hunk 2]
```diff
@@ -29,48 +30,49 @@ import (
...
+		servingLister:    serviceInformer.Lister(),
+		servingClientSet: serviceclient.Get(ctx),
+		gitlabClientSet:  gitlabclient.Get(ctx),
+		gitlabLister:     gitlabInformer.Lister(),
+		loggingContext:   ctx,
+	}
+
+	env := &envConfig{}
+	if err := envconfig.Process("", env); err != nil {
+		logging.FromContext(ctx).Panicf("unable to process PingSourceSource's required environment variables: %v", err)
```
Comment (odacremolbap) [inline gitlab/pkg/reconciler/controller.go]: Not `PingSource`

File: gitlab/pkg/reconciler/gitlabsource.go [hunk 3]
```diff
@@ -233,7 +149,7 @@ func (r *Reconciler) reconcile(source *sourcesv1alpha1.GitLabSource) error {
 		}
 	}
 
-	ksvc, err = r.waitForKnativeServiceReady(source)
+	ksvc, err = r.waitForKnativeServiceReady(source) // TODO: remove this. setup a watch on the service instead.
```
Comment (vaikas) [inline gitlab/pkg/reconciler/gitlabsource.go]: Can we create issues to track this and the other TODO?

Comment (n3wscott) [reply gitlab/pkg/reconciler/gitlabsource.go]: https://github.com/knative/eventing-contrib/issues/1066

File: gitlab/pkg/reconciler/gitlabsource.go [hunk 4]
```diff
@@ -409,7 +305,7 @@ func (r *Reconciler) getOwnedKnativeService(source *sourcesv1alpha1.GitLabSource
 }
 
 // UpdateFromLoggingConfigMap loads logger configuration from configmap
-func (r *Reconciler) UpdateFromLoggingConfigMap(cfg *corev1.ConfigMap) {
+func (r *Reconciler) UpdateFromLoggingConfigMap(cfg *corev1.ConfigMap) { // TODO: remove this.
```
Comment (n3wscott) [inline gitlab/pkg/reconciler/gitlabsource.go]: This is core functionality in other places and should be handled at the controller level, not the reconciler. See eventing and serving on how to do this.

**Pull request merged**